trk/src/Components/SessionDetail.js

```diff
@@ -13,46 +13,91 @@ import { ActivityIndicator } from "react-native-paper";
+    return;
+  }
   const title = route.params.title;
   //data[0].images = [{path: 'climb photos/the_crag.png'},{path: 'climb photos/the_crag.png'},{path: 'climb photos/the_crag.png'},{path: 'climb photos/the_crag.png'}];
   const allImages = useMemo(() => {
     return data.flatMap(item => item.sessionImages ? item.sessionImages : []);
   }, [data]);  
-  console.log('All Images: ', allImages);
+  //console.log('All Images: ', allImages);
   let initallySelected = null;
   let initialText = null;
   let selectedData = null;
-  console.log(data);
+  //console.log(data);
   //Selected Item Set
   if (data.length>0) {
       selectedData = data.filter(obj => obj.isSelected == true);
-      console.log('The selected Data is: ', selectedData);
+      //console.log('The selected Data is: ', selectedData);
       if (selectedData.length == 0) {
           initallySelected = data[0].tapId;
           selectedData = [data[0]];
-          console.log('Initially selected value is: ', initallySelected);
+          //console.log('Initially selected value is: ', initallySelected);
       } else {
           initallySelected = selectedData[0].tapId;
-          console.log('Initially selected value is: ', initallySelected);
+          //console.log('Initially selected value is: ', initallySelected);
       }
       if (data[data.length-1].sessionTitle === undefined || (data[data.length-1].sessionTitle && data[data.length-1].sessionTitle === '')) {
           initialText = 'Session on '+title[1];
       }
       else {
           initialText = data[data.length-1].sessionTitle;
       }
-      console.log('Initial session text: ', initialText);
+      //console.log('Initial session text: ', initialText);
   }
 
   const [climbImageUrl, setClimbImageUrl] = useState(null);
   const { currentUser } = useContext(AuthContext);
 
   const [selectedImage, setSelectedImage] = useState(null);
   const [modalVisible, setModalVisible] = useState(false);
+  let tagged = (data[data.length-1].tagged? data[data.length-1].tagged: []);
+  const [taggedWithImages, setTaggedWithImages] = useState(null);
+
+  const handleImageFetch = async () => {
+    let combinedUsers = [];
+    for (const searchQuery of tagged) {
+        // Assuming you have separate API functions for fetching by username and email
+        const querySnapshotByUsername = await UsersApi().getUsersByForSearch(searchQuery);
+        const usersByUsername = querySnapshotByUsername.docs.map(doc => doc.data());
+        const querySnapshotByEmail = await UsersApi().getUsersByForSearchEmail(searchQuery);
+        const usersByEmail = querySnapshotByEmail.docs.map(doc => doc.data());
+        // Combine and deduplicate users
+        combinedUsers = [...combinedUsers, ...usersByUsername, ...usersByEmail];
+    }
+    let uniqueUsers = Array.from(new Set(combinedUsers.map(user => user.uid)))
+        .map(uid => {
+            return combinedUsers.find(user => user.uid === uid);
```

Comment:
Is this to remove duplicates when email = username?

Reply:
Yupp because for users that haven't set a username, we use their email Ids instead- the part before the '@'. The 2 queries may then fetch duplicate records.